### PR TITLE
add a global check (15 minutes) and use the current limits for specif…

### PR DESCRIPTION
**Pull request by garrettlangley**
…ic geofences

### CODE REVIEW

File: src/ios/GeofencePlugin.swift [hunk 1]
```diff
@@ -586,40 +592,52 @@ class GeoNotificationManager : NSObject, CLLocationManagerDelegate {
...
         let date = convertDateToString(NSDate())
-        let labelTime = "moment_push_time"
+        let labelTime = "mo-" + key
         defaults.setObject(date, forKey: labelTime)
         NSUserDefaults.standardUserDefaults().synchronize()
     }
 
     func checkRateLimit(geo: JSON) -> Bool {
-        let previousDate = getNotificationTimestamp()
+        let geoId = geo["id"].stringValue
```
Comment (jnystrom) [inline src/ios/GeofencePlugin.swift]: is this for checking the rate limit for the specific fence, and not the global one?

File: src/ios/GeofencePlugin.swift [hunk 2]
```diff
@@ -535,7 +536,8 @@ class GeoNotificationManager : NSObject, CLLocationManagerDelegate {
                     if checkRateLimit(geoNotification) == true {
                         notifyAbout(geoNotification)
                         notifyServer(geoNotification, location: location)
-                        setNotificationTimestamp()
+                        setNotificationTimestamp(geoNotification["id"].stringValue)
+                        setNotificationTimestamp("global")
```
Comment (jnystrom) [inline src/ios/GeofencePlugin.swift]: this just looks weird.  At least some comments.

File: src/ios/GeofencePlugin.swift [hunk 3]
```diff
@@ -586,40 +592,52 @@ class GeoNotificationManager : NSObject, CLLocationManagerDelegate {
...
 
         if let customRateLimit = geo["notification"]["data"]["frequency"].int {
-            rateLimit = customRateLimit
+            rateLimitForGeofence = customRateLimit
         }
-
-        if previousDate == "empty" {
+        log("Looking at dates = \(previousGeofenceDate), \(previousGlobalDate)")
+        if previousGeofenceDate == "empty" || previousGlobalDate == "empty" {
             return true
```
Comment (jnystrom) [inline src/ios/GeofencePlugin.swift]: why are you checking for the string "empty"?

**Comment by jnystrom** [comment]
I did not dig to deep into the logic, I assume it was tested.  I just tried to point out somethings I just found confusing.

**Comment by garrettlangley** [comment]
@jnystrom thanks, i added comments in places you had questions.

**Pull request merged**